presto-parquet/src/main/java/io/prestosql/parquet/ParquetTimestampUtils.java

```diff
@@ -54,11 +57,29 @@ public static long getTimestampMillis(Binary timestampBinary)
         long timeOfDayNanos = Longs.fromBytes(bytes[7], bytes[6], bytes[5], bytes[4], bytes[3], bytes[2], bytes[1], bytes[0]);
         int julianDay = Ints.fromBytes(bytes[11], bytes[10], bytes[9], bytes[8]);
 
-        return julianDayToMillis(julianDay) + (timeOfDayNanos / NANOS_PER_MILLISECOND);
+        long epochSeconds = (julianDay - JULIAN_EPOCH_OFFSET_DAYS) * SECONDS_PER_DAY + timeOfDayNanos / NANOS_PER_SECOND;
+        return new DecodedTimestamp(epochSeconds, (int) (timeOfDayNanos % NANOS_PER_SECOND));
```

Comment:
`verify(0 <= timeOfDayNanos <= 86400000000000)`?
(if `timeOfDayNanos` can be negative, the code would not be correct)